src/applications/disability-benefits/2346/schemas/address-schema.js

```diff
@@ -0,0 +1,258 @@
+ * 2. USA - used to just reference the United States
+ * 3. MilitaryBaseInfo - expandable text to expound on military base addresses.
+ * 4. addressSchema - data model for address schema.
+ */
+
+const MILITARY_STATES = Object.entries(ADDRESS_DATA.states).reduce(
+  (militaryStates, [stateCode, stateName]) => {
+    if (ADDRESS_DATA.militaryStates.includes(stateCode)) {
+      return {
+        ...militaryStates,
+        [stateCode]: stateName,
+      };
+    }
+    return militaryStates;
+  },
+  {},
+);
+
+const USA = {
+  value: 'USA',
+  name: 'United States',
+};
+
+const MilitaryBaseInfo = () => (
+  <div className="vads-u-padding-x--2p5">
+    <AdditionalInfo
+      status="info"
+      triggerText="Learn more about military base addresses"
+    >
+      <span>
+        The United States is automatically chosen as your country if you live on
+        a military base outside of the country.
+      </span>
+    </AdditionalInfo>
+  </div>
+);
+
+const addressSchema = {
+  type: 'object',
+  properties: {
+    'view:livesOnMilitaryBase': {
+      type: 'boolean',
+    },
+    'view:livesOnMilitaryBaseInfo': {
+      type: 'object',
+      properties: {},
+    },
+    country: {
+      type: 'string',
+      enum: countries.map(country => country.value),
+    },
+    street: {
+      type: 'string',
+      minLength: 1,
+      maxLength: 100,
+      pattern: '^.*\\S.*',
+    },
+    street2: {
+      type: 'string',
+      minLength: 1,
+      maxLength: 100,
+      pattern: '^.*\\S.*',
+    },
+    city: {
+      type: 'string',
+    },
+    state: {
+      type: 'string',
+      enum: states50AndDC.map(state => state.value),
+      enumNames: states50AndDC.map(state => state.label),
+    },
+    postalCode: {
+      type: 'string',
+      pattern: '^\\d{5}$',
+    },
+  },
+};
+
+/**
+ * Builds address schema based on isMilitaryAddress.
+ * @param {boolean} isMilitaryBaseAddress represents whether or not the form page requires the address to support the option of military address.
+ * @returns {object} an object containing the necessary properties for a domestic US address, foreign US military address, and international address.
+ */
+export const buildAddressSchema = isMilitaryBaseAddress => {
+  if (isMilitaryBaseAddress) return cloneDeep(addressSchema);
+  const schema = cloneDeep(addressSchema);
+  delete schema.properties['view:livesOnMilitaryBase'];
+  delete schema.properties['view:livesOnMilitaryBaseInfo'];
+  return schema;
+};
+
+/**
+ * This method takes a list of parameters and generates an addressUiSchema.
+ * @param {function} callback slots into the 'ui:required' for the necessary fields.
+ * @param {string} path represents the path to the address in formData.
+ * @param {boolean} isMilitaryBaseAddress represents whether or not the form page requires the address to support the option of military address.
+ */
+
+const MILITARY_BASE_PATH = '[view:livesOnMilitaryBase]';
```

Comment:
Hmm, did we mean to make this a stringified array here? 🙂

Reply:
Maybe? 😂 Ill find out